File: string/two_pointer/reverse_a_string.js
function reverse(str){
    let chars = str.split("");
    let left =0;
    let right = chars.length -1
    while(left<right){
        [chars[right],chars[left]] =[chars[left],chars[right]]
        left++;
        right--;
    }
    return chars.join("")
}


console.log(reverse("hello"));

File: string/two_pointer/reverse_a_string.py

def reverse(str:str):
    chars = list(str)
    left =0
    right= len(chars)-1
    while(left<right):
        chars[right], chars[left] = chars[left], chars[right]
        left +=1
        right -=1
    return "".join(chars)

print(reverse("hello"))